examples/rvc-app/rvc-common/src/rvc-service-area-delegate.cpp

```diff
@@ -0,0 +1,467 @@
+{
+    bool ret_value = false;
+
+    if (listIndex < (uint32_t) mSupportedLocations.size())
+    {
+        aSupportedLocation = mSupportedLocations[listIndex];
+        ret_value          = true;
+    }
+
+    return ret_value;
+};
+
+bool RvcServiceAreaDelegate::GetSupportedLocationById(uint32_t aLocationId, uint32_t & listIndex,
+                                                      LocationStructureWrapper & aSupportedLocation)
+{
+    // We do not need to reimplement this method as it's already done by the SDK.
+    // We are reimplementing this method, still using linear search, but with some optimization on the SDK implementation
+    // since we have direct access to the list.
+    bool ret_value = false;
+
+    listIndex         = 0;
+    uint32_t listSize = (uint32_t) mSupportedLocations.size();
+
+    while (listIndex < listSize)
+    {
+        if (mSupportedLocations[listIndex].locationID == aLocationId)
+        {
+            aSupportedLocation = mSupportedLocations[listIndex];
+            ret_value          = true;
+            break;
+        }
+
+        ++listIndex;
+    }
+
+    return ret_value;
+};
+
+bool RvcServiceAreaDelegate::AddSupportedLocation(const LocationStructureWrapper & newLocation, uint32_t & listIndex)
+{
+    bool ret_value = false;
+
+    // The server instance (caller) is responsible for ensuring that there are no duplicate location IDs, list size not exceeded,
+    // etc.
+
+    // Double-check list size to ensure there no memory issues.
+    if ((uint32_t) mSupportedLocations.size() < kMaxNumSupportedLocations)
+    {
+        // not sorting list, number of locations normally expected to be small, max 255
+        mSupportedLocations.push_back(newLocation);
+        listIndex = (uint32_t) mSupportedMaps.size() - 1; // new element is last in list
+        ret_value = true;
+    }
+    else
+    {
+        ChipLogError(Zcl, "AddSupportedLocation %u - supported locations list is already at maximum size %u",
+                     newLocation.locationID, (uint32_t) kMaxNumSupportedLocations);
+    }
+
+    return ret_value;
+}
+
+bool RvcServiceAreaDelegate::ModifySupportedLocation(uint32_t listIndex, const LocationStructureWrapper & modifiedLocation)
+{
+    bool ret_value = false;
+
+    // The server instance (caller) is responsible for ensuring that there are no duplicate location IDs, list size not exceeded,
+    // etc.
+
+    // Double-check that locationID's match.
+    VerifyOrExit((modifiedLocation.locationID == mSupportedLocations[listIndex].locationID),
+                 ChipLogError(Zcl, "ModifySupportedLocation - new locationID %u does not match existing locationID %u",
+                              modifiedLocation.locationID, mSupportedLocations[listIndex].locationID));
+
+    // checks passed, update the attribute
+    mSupportedLocations[listIndex] = modifiedLocation;
+    ret_value                      = true;
+
+exit:
```

Comment:
please avoid goto-exit usage in new code.

Reply:
This has been addressed in [this commit](https://github.com/project-chip/connectedhomeip/pull/33991/commits/38a7e7a9c206391ede4528ffd1bc73c391c400a1).